pkg/oc/cli/cmd/tag.go

```diff
@@ -145,6 +145,15 @@ func determineSourceKind(f *clientcmd.Factory, input string) string {
 		return "DockerImage"
 	}
 
+	glog.V(1).Infof("error while determining source kind: %v", err)
+
+	switch strings.ToLower(input) {
```

Comment:
We should not need this. The information about the aliases is read by the mapper and should be properly resolved in line 137 above here. The only reason we have the dockerimage exception is that this particular type is not registered in restmapper and will never be. In all other cases the resolution should happen. What's your particular use case in, why this is happening?

Reply:
I just tried current master and I got:
```
$ oc tag iknow:wrong foo:bar --source=isimage --loglevel=3
I0130 11:24:43.397672    2223 tag.go:179] Using "iknow:wrong" as a source tag
F0130 11:24:43.472065    2223 helpers.go:119] error: --source=ImageStreamImage requires a valid <name>@<id> in SOURCE
```

Reply:
> What's your particular use case in, why this is happening?

The error I reported here could happen with version mismatch or server's configuration (Although I haven't researched the reason yet, the command should handle the error in the first place).

I think that you can produce the error if you use https://api.ded-stage-aws.openshift.com:443 and the latest command binary. At least I am getting the error on my env.

~~~
$ _output/local/bin/linux/amd64/oc version
oc v1.0.0+9ab7a71-18788
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://api.ded-stage-aws.openshift.com:443
openshift v3.6.173.0.96
kubernetes v1.6.1+5115d708d7
~~~

Reply:
If the code relies on the restmapper, `determineSourceKind` should returns result and error excactly and should NOT return `input` arg. (But as I commented, the latest binary and some(old?) cluster fails to get `istag` and `isimage`, so I think it should handle as my current PR.)

~~~
func determineSourceKind(f *clientcmd.Factory, input string) (string, eror) {
	// DockerImage isn't in RESTMapper
	switch strings.ToLower(input) {
	case "docker", "dockerimage":
		return "DockerImage", nil
	}

	mapper, _ := f.Object()
	gvks, err := mapper.KindsFor(schema.GroupVersionResource{Group: imageapi.GroupName, Resource: input})
	return gvks[0].Kind, err
}
~~~

Reply:
Sorry for being late, the general rule of thumb is that we support n-1 compatibility for oc. Additionally, 3.6 does not publish the discovery information we currently rely on, and hard coding fallback is not an option here.

Reply:
@soltysh Thank you. I understand. Then, I updated `determineSourceKind()` not to return `input` arg when REST mapper failed, as it behaves unexpectedly.